internal/rego/load.go

```diff
@@ -0,0 +1,116 @@
+package rego
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/open-policy-agent/opa/ast"
+)
+
+// LoadPoliciesWithAction loads all policies from rego with rules with a given action name
+func LoadPoliciesWithAction(filesContents map[string]string, action string) ([]File, error) {
```

Comment:
To make the API call easier for consumers, I would also move the `readFilesContents` over to the rego library, but private, and have the public functions take paths, e.g. GetPolicies(path).

It also may make more sense to have separate Libray/Policy types, though I'm not really sure at the moment.

Reply:
This requires making `newRegoFile()` public in order to test the create and doc commands without writing temp files to disk, but  I'm OK with that.